sklearn/metrics/cluster/tests/test_unsupervised.py

```diff
@@ -61,14 +61,75 @@ def test_correct_labelsize():
     y = np.arange(X.shape[0])
     assert_raises_regexp(ValueError,
                          "Number of labels is %d "
-                         "but should be more than 2"
-                         "and less than n_samples - 1" % len(np.unique(y)),
+                         "but should be at least 2 "
+                         "and less than n_samples" % len(np.unique(y)),
                          silhouette_score, X, y)
 
     # n_labels = 1
     y = np.zeros(X.shape[0])
     assert_raises_regexp(ValueError,
                          "Number of labels is %d "
-                         "but should be more than 2"
-                         "and less than n_samples - 1" % len(np.unique(y)),
+                         "but should be at least 2 "
+                         "and less than n_samples" % len(np.unique(y)),
                          silhouette_score, X, y)
+
+
+def test_paper_example():
```

Comment:
Can you maybe add the reference Silhouettes: a Graphical Aid to the Interpretation and Validation of Cluster Analysis again?